### DISCUSSION TITLE
onComplete event

### DISCUSSION THREAD

**Discussion by arnaudjnn** [urql-graphql/urql#2233] [Q&A] [answered]
I need to get the data length in my parent component, how can I do? Why there is no `onComplet` callback?

**Comment by kimitaWanjohi** [comment]
Yeah Why is there no onComplete?

**Comment by kitten** [comment, marked as answer]
There's a short explanation here: https://github.com/FormidableLabs/urql/issues/1449
But the short answer is that they're really not needed and lead often to poorly designed interactions, since either effects or the promise-based APIs are more appropriate, depending on the use-case.

I'm not sure what you mean by data length, but, in React at least, I'd never try to send simple data subsets back to the parent. It's not idiomatic and leads to more issues than it's worth. If you need the data there for a simple UI element I'd maybe consider hoisting the `useQuery` hook or restructuring your markup ✌️
